### ISSUE TITLE
aws_emr_cluster enhancement to support KerberosAttributes

### ISSUE THREAD

**Issue by FireballDWF** [terraform-providers/terraform-provider-aws#3477]
### Terraform Version
0.10.8
aws_provider 1.9.0

### Affected Resource(s)
- aws_emr_cluster

### Expected Behavior
Support the following attributes of the RunJobFlow API documented at https://docs.aws.amazon.com/emr/latest/APIReference/API_KerberosAttributes.html
```
   "KerberosAttributes": { 
      "ADDomainJoinPassword": "string",
      "ADDomainJoinUser": "string",
      "CrossRealmTrustPrincipalPassword": "string",
      "KdcAdminPassword": "string",
      "Realm": "string"
   }
```

**Comment by bflad** [comment]
@FireballDWF have you taken a look at using the [`aws_emr_security_configuration` resource](https://www.terraform.io/docs/providers/aws/r/emr_security_configuration.html) to achieve this? 

Something like the following should work using an example configuration from the [AWS documentation](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-kerberos-configure.html):

```hcl
resource "aws_emr_security_configuration" "kerberos-example" {
  name = "kerberos-example"

  configuration = <<EOF
{
  "AuthenticationConfiguration": {
    "KerberosConfiguration": {
      "Provider": "ClusterDedicatedKdc",
      "ClusterDedicatedKdcConfiguration": {
        "TicketLifetimeInHours": 24,
        "CrossRealmTrustConfiguration": {
          "Realm": "AD.DOMAIN.COM",
          "Domain": "ad.domain.com",
          "AdminServer": "ad.domain.com",
          "KdcServer": "ad.domain.com"
        }
      }
    }
  }
}
EOF
}

resource "aws_emr_cluster" "kerberos-example" {
  # ... other configuration ...
  security_configuration = "${aws_emr_security_configuration.kerberos-example.}"
}
```

If the above works for your situation, maybe we can close this with the additional Kerberos example in the `aws_emr_security_configuration` resource?

**Comment by FireballDWF** [comment]
When KerberosConfiguration attributes in a SecurityConfiguration passed to RunJobFlow api are present, then the kerberos attributes are required per Step 6.2 of https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-kerberos-cross-realm.html#emr-
kerberos-ad-cluster
Think about it, how else are passwords defined to the cluster when you are configuring Authorization?  There are no passwords attributes in SecurityConfiguration.

**Comment by bflad** [comment]
Sorry I had missed the distinction between configuring the passwords and just the baseline Kerberos settings in the security configuration. I'll work on adding those to the cluster resource.

**Comment by bflad** [comment]
PR submitted: #3553

**Comment by bflad** [comment]
Support has been merged into master and will be released with v1.11.0 of the AWS provider, likely at the end of the week.

**Comment by bflad** [comment]
This has been released in [version 1.11.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1110-march-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by FireballDWF** [comment]
I tested it yesterday and it worked fine, thank you!
